pkg/search/postgres/store.go

```diff
@@ -349,3 +350,30 @@ func (s *GenericStore[T, PT]) GetMany(ctx context.Context, identifiers []string)
 	}
 	return elems, missingIndices, nil
 }
+
+// DeleteByQuery removes the objects from the store based on the passed query.
+func (s *GenericStore[T, PT]) DeleteByQuery(ctx context.Context, query *v1.Query) error {
+	defer s.setPostgresOperationDurationTime(time.Now(), ops.Remove)
```

Comment:
While not directly linked to the change at hand, it might be good to differentiate `Delete` from `DeleteByQuery` in the stat histograms. If the filter of `DeleteByQuery` is complex enough, this would introduce false outliers in the distribution.

Reply:
Ideally we should track time and length of the query together so we can see correlation. Anyway I think it might be hard after this simplification: https://github.com/stackrox/stackrox/pull/6775#discussion_r1254670891